packages/data-api/src/TupaiaDataApi.ts

```diff
@@ -188,25 +214,37 @@ export class TupaiaDataApi {
     return groupBy(options, 'option_set_id');
   }
 
-  buildOptionsMetadata = (options = [], type) => {
+  private buildOptionsMetadata = (
+    options: (string | { value: string; label: string | undefined })[] = [],
+    type: string,
+  ) => {
     const optionList = options.length === 0 && type === 'Binary' ? DEFAULT_BINARY_OPTIONS : options;
 
     if (!optionList.length) {
       return undefined;
     }
 
-    const optionsMetadata = {};
+    const optionsMetadata: Record<string, string> = {};
+    const optionsObjectValidator = yup.object({
+      value: yup.string().required(),
+      label: yup.string(),
+    });
 
     optionList.forEach(option => {
-      try {
-        // options coming from question.options are JSON format strings
-        // options coming from option_set are actual JSON objects
-        const optionObject = typeof option === 'string' ? JSON.parse(option) : option;
-        const { value, label } = optionObject;
+      if (typeof option === 'string') {
+        try {
+          // options coming from question.options are JSON format strings
+          // options coming from option_set are actual JSON objects
+          const optionObject = optionsObjectValidator.validateSync(JSON.parse(option));
```

Comment:
Still feel a bit awkward that this logic uses exception catching like it does... And technically the behaviour has slightly changed by using a validator here as it will throw an exception if the expected fields aren't present... Might rethink this...

Reply:
Decided to leave it for now as it feels appropriate... happy to hear reviewer's thoughts though 👍

Reply:
Yeah I don't really get the throw-catch logic... maybe pop your thinking into the code for a future cleanup, otherwise happy with it.